internal/controller/volsync/vshandler_test.go

```diff
@@ -676,6 +676,51 @@ var _ = Describe("VolSync_Handler", func() {
 
 					Expect(pvc.GetName()).To(Equal(rdSpec.ProtectedPVC.Name))
 					Expect(pvc.GetOwnerReferences()[0].Kind).To(Equal("ConfigMap"))
+					Expect(k8sClient.Delete(ctx, pvc)).ToNot(HaveOccurred())
+				})
+
+				It("PrecreateDestPVCIfEnabled() should create PVC with volumeMode: Filesystem", func() {
+					dstPVC, err := vsHandler.PrecreateDestPVCIfEnabled(rdSpec)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(*dstPVC).To(Equal(rdSpec.ProtectedPVC.Name))
+					pvc := &corev1.PersistentVolumeClaim{}
+					Eventually(func() error {
+						return k8sClient.Get(ctx, types.NamespacedName{
+							Name:      rdSpec.ProtectedPVC.Name,
+							Namespace: testNamespace.GetName(),
+						}, pvc)
+					}, maxWait, interval).Should(Succeed())
+
+					Expect(pvc.GetName()).To(Equal(rdSpec.ProtectedPVC.Name))
+					Expect(pvc.GetOwnerReferences()[0].Kind).To(Equal("ConfigMap"))
+					Expect(*pvc.Spec.VolumeMode).To(Equal(corev1.PersistentVolumeFilesystem))
+					Expect(k8sClient.Delete(ctx, pvc)).ToNot(HaveOccurred())
+				})
+
+				It("PrecreateDestPVCIfEnabled() should create PVC with volumeMode: Block", func() {
+					util.AddAnnotation(owner, util.UseVolSyncForRBDAnnotation, "true")
+					Expect(k8sClient.Update(ctx, owner)).To(Succeed())
+					rdSpecForBlockPVC := rdSpec.DeepCopy()
+					block := corev1.PersistentVolumeBlock
+					rdSpecForBlockPVC.ProtectedPVC.VolumeMode = &block
+
+					dstPVC, err := vsHandler.PrecreateDestPVCIfEnabled(*rdSpecForBlockPVC)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(*dstPVC).To(Equal(rdSpecForBlockPVC.ProtectedPVC.Name))
+					pvc := &corev1.PersistentVolumeClaim{}
+					Eventually(func() error {
+						return k8sClient.Get(ctx, types.NamespacedName{
+							Name:      rdSpecForBlockPVC.ProtectedPVC.Name,
+							Namespace: testNamespace.GetName(),
+						}, pvc)
+					}, maxWait, interval).Should(Succeed())
+
+					Expect(pvc.GetName()).To(Equal(rdSpecForBlockPVC.ProtectedPVC.Name))
+					Expect(pvc.GetOwnerReferences()[0].Kind).To(Equal("ConfigMap"))
+					Expect(*pvc.Spec.VolumeMode).To(Equal(corev1.PersistentVolumeBlock))
+					Expect(k8sClient.Delete(ctx, pvc)).ToNot(HaveOccurred())
```

Comment:
This is a duplicate of the previous test - we have different setup and one different assertion. Can extract the repeated test code to a helper?

Different setup:

```go
util.AddAnnotation(owner, util.UseVolSyncForRBDAnnotation, "true")
Expect(k8sClient.Update(ctx, owner)).To(Succeed())
rdSpecForBlockPVC := rdSpec.DeepCopy()
block := corev1.PersistentVolumeBlock
rdSpecForBlockPVC.ProtectedPVC.VolumeMode = &block
```

Different check:

```go
Expect(*pvc.Spec.VolumeMode).To(Equal(corev1.PersistentVolumeBlock))
```

Reply:
Sure. I'll extract it to a helper function.

Reply:
Did you miss this comment?